modules/sdk-lib-mpc/src/tss/ecdsa/paillierBlumProof.ts

```diff
@@ -0,0 +1,134 @@
+import { createHmac } from 'crypto';
+import { bitLength, randBits, isProbablyPrime } from 'bigint-crypto-utils';
+import { modInv, modPow } from 'bigint-mod-arith';
+import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
+import { DeserializedPaillierBlumProof } from './types';
+
+// Security parameter.
+const m = 80;
+
+/**
+ * Generate psuedo-random quadratic residue for (N, w, i).
+ * @param N - the prime number to verify is a product of two large primes.
+ * @param w - a random number with the same bitLength as N, that satisfies the Jacobi of w is -1 wrt N.
+ * @returns {bigint[]} - set of challenges for N
+ */
+function generateY(N, w): bigint[] {
```

Comment:
Im not sure where in this function does rejection sampling is done with the set $$Z^{*}_{N}$$ in mind. Need to do gcd(y,N) = 1 to check membership as described in Nothing-up-my-sleeve constructions spec here https://www.zkdocs.com/docs/zkdocs/protocol-primitives/nums/.

Reply:
cc @johnoliverdriscoll

Reply:
Done